core/middlewares/with-makeswift.ts

```diff
@@ -0,0 +1,39 @@
+import { NextRequest } from 'next/server';
+import { parse as parseSetCookie } from 'set-cookie-parser';
+
+import { MiddlewareFactory } from './compose-middlewares';
+
+export const withMakeswift: MiddlewareFactory = (middleware) => {
+  return async (request, event) => {
+    const apiKey = request.nextUrl.searchParams.get('x-makeswift-draft-mode');
```

Comment:
```suggestion
    const apiKey = request.nextUrl.searchParams.get('x-makeswift-draft-mode') ?? request.headers.get('x-makeswift-draft-mode');
```

This change is necessary for a "proper" proxy implementation. @agurtovoy, I wonder if this has anything to do with server actions misbehaving in Catalyst